vcd/resource_vcd_vapp_vm.go

```diff
@@ -241,6 +242,104 @@ func resourceVcdVAppVm() *schema.Resource {
 				Optional: true,
 				Set:      resourceVcdVmIndependentDiskHash,
 			},
+			"override_template_disk": {
+				Type:        schema.TypeSet,
+				Optional:    true,
+				ForceNew:    true,
+				Description: " A block to match internal_disk interface in template. Multiple can be used. Disk will be matched by bus_type, bus_number and unit_number.",
+				Elem: &schema.Resource{Schema: map[string]*schema.Schema{
+					"bus_type": {
+						Type:         schema.TypeString,
+						Required:     true,
+						ForceNew:     true,
+						ValidateFunc: validation.StringInSlice([]string{"ide", "parallel", "sas", "paravirtual", "sata"}, true),
+						Description:  "The type of disk controller. Possible values: ide, parallel( LSI Logic Parallel SCSI), sas(LSI Logic SAS (SCSI)), paravirtual(Paravirtual (SCSI)), sata",
+					},
+					"size_in_mb": {
+						Type:        schema.TypeInt,
+						ForceNew:    true,
+						Required:    true,
+						Description: "The size of the disk in MB.",
+					},
+					"bus_number": {
+						Type:        schema.TypeInt,
+						ForceNew:    true,
+						Required:    true,
+						Description: "The number of the SCSI or IDE controller itself.",
+					},
+					"unit_number": {
+						Type:        schema.TypeInt,
+						ForceNew:    true,
+						Required:    true,
+						Description: "The device number on the SCSI or IDE controller of the disk.",
+					},
+					"thin_provisioned": {
+						Type:        schema.TypeBool,
+						ForceNew:    true,
+						Optional:    true,
+						Description: "Specifies whether the disk storage is pre-allocated or allocated on demand.",
+					},
+					"iops": {
+						Type:        schema.TypeInt,
+						ForceNew:    true,
+						Optional:    true,
+						Description: "Specifies the IOPS for the disk. Default - 0.",
+					},
+					"storage_profile": &schema.Schema{
+						Type:        schema.TypeString,
+						ForceNew:    true,
+						Optional:    true,
+						Description: "Storage profile to override the VM default one",
+					},
+				}},
+			},
+			"internal_disk": {
+				Type:        schema.TypeList,
```

Comment:
How is it sorted - does it concide with UI Hard disk numbers?

Reply:
This whole part is computed and just dumps what it gets from API, always overwrite. No sorting or matching.